java-checks-test-sources/src/main/java/checks/TypeParametersShadowingCheck.java

```diff
@@ -0,0 +1,108 @@
+package checks;
+
+import java.util.List;
+import java.util.function.Function;
+
+public class TypeParametersShadowingCheck<T0> {
+  class TypeParameterHidesAnotherType<T> {
+    public class Inner<T> { // Noncompliant [[sc=24;ec=25;secondary=-1]] {{Rename "T" which hides a type parameter from the outer scope.}}
+      //...
+    }

```

Comment:
I'd like a second inner class here that shadows T0 instead of T (`public class Inner2<T0> {}`) to test the fact that we don't forget about `TypeParametersShadowingCheck`'s type parameters when entering `TypeParameterHidesAnotherType`.